File: cuckoo-hibersap-example-datamodel/src/main/java/de/akquinet/jbosscc/cuckoo/example/conversion/SeverityConverter.java
package de.akquinet.jbosscc.cuckoo.example.conversion;

import de.akquinet.jbosscc.cuckoo.example.model.Severity;
import org.hibersap.conversion.ConversionException;
import org.hibersap.conversion.Converter;

public class SeverityConverter implements Converter
{
    public Severity convertToJava( Object sapValue ) throws ConversionException
    {
        String sapType = ( String ) sapValue;
        return Severity.fromSapType( sapType.charAt( 0 ) );
    }

    public String convertToSap( Object javaValue ) throws ConversionException
    {
        throw new UnsupportedOperationException( "The parameter should never be passed back to SAP" );
    }
}

File: src/main/kotlin/org/hibersap/customer/conversion/SeverityConverter.kt
package org.hibersap.customer.conversion

import org.hibersap.conversion.Converter
import org.hibersap.customer.model.Severity
import org.hibersap.customer.model.Severity.Companion.fromSapType

class SeverityConverter : Converter<Severity, String> {

    override fun convertToJava(sapValue: String): Severity {
        return fromSapType(sapValue[0])
    }

    override fun convertToSap(javaValue: Severity): String {
        return javaValue.sapType.toString()
    }
}